packages/block-editor/src/components/block-list/use-block-props/use-block-refs.js

```diff
@@ -63,7 +66,17 @@ function useBlockRef( clientId ) {
 	return useMemo(
 		() => ( {
 			get current() {
-				return refs.get( freshClientId.current )?.current || null;
+				let element = null;
+
+				// Multiple refs may be created for a single block. Find the
+				// first that has an element set.
+				for ( const [ ref, id ] of refs.entries() ) {
+					if ( id === freshClientId.current && ref.current ) {
```

Comment:
I'm not sure about this one... You've changed the key to have `refs` but the check here remains with `clientId`. Is the first one the right one though or we need the last one? 🤔

Reply:
It shouldn't matter because only one will be set. IF multiple ones are set somehow, they are the same element.